File: CrudLayered_VB/DataAccess/Repositories/Repository.vb
Imports System.Configuration
Imports MySql.Data.MySqlClient


Public MustInherit Class Repository
    Private ReadOnly connectionString As String

    Public Sub New()
        connectionString = ConfigurationManager.ConnectionStrings("ConnMyCompany").ToString()
    End Sub

    Protected Function GetConnection() As MySqlConnection
        Dim conexion As MySqlConnection = New MySqlConnection
        conexion.ConnectionString = connectionString
        Return conexion
    End Function
End Class

File: Proyecto/Proyecto.Data/Repositorios/Repository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Configuration;
using System.Data.SqlClient;

namespace Proyecto.Data.Repositorios
{
    public abstract class Repository
    {
        private readonly string connectionString;

        public Repository()
        {
            connectionString = ConfigurationManager.ConnectionStrings["connMyCompany"].ToString();
        }

        protected SqlConnection GetConnection()
        {
            return new SqlConnection(connectionString);
        }
    }
}
